java/ql/src/experimental/Security/CWE/CWE-117/LogInjection.java

```diff
@@ -0,0 +1,34 @@
+package com.example.restservice;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class LogInjection {
+
+    private final Logger log = LoggerFactory.getLogger(LogInjection.class);
+
+    // /good?username=Guest'%0AUser:'Admin
+    @GetMapping("/good")
+    public String good(@RequestParam(value = "username", defaultValue = "name") String username) {
+        if (username.matches("[A-z1-9]")) {
```

Comment:
```suggestion
        if (username.matches("\w*")) {
```
`[A-z1-9]` only matches a single character, and it includes the characters between `Z` and `a` (such as `[`). I suggest using `\w`, which is equivalent to `[a-zA-Z_0-9]`.